app/src/lib/lookups/form.js

```diff
@@ -1,73 +1,64 @@
-			return
-		}
+	if (lookup_results.length === 0) {
+		return
+	}
 
-		// The form lookup may have resulted in different stems (eg. saw).
-		// So add a lookup term for each unique stem (case-insensitive)
-		const unique_stems = new Set(lookup_results.map(({ stem }) => stem.toLowerCase()))
+	// The form lookup may have resulted in different stems (eg. saw).
+	// So add a lookup term for each unique stem (case-insensitive)
+	const unique_stems = new Set(lookup_results.map(({ stem }) => stem.toLowerCase()))
 
-		// Exclude the original lookup stem as it's handled separately
-		unique_stems.delete(term.toLowerCase())
+	// Exclude the original lookup stem as it's handled separately
+	unique_stems.delete(term.toLowerCase())
 
-		// Keep the original token-based term as the first lookup term
-		token.lookup_terms = [term.toLowerCase(), ...unique_stems]
-		token.lookup_results = lookup_results
-	}
+	// Keep the original token-based term as the first lookup term
+	lookup_token.lookup_terms = [term.toLowerCase(), ...unique_stems]
+	lookup_token.lookup_results = lookup_results
 
 	/**
+	 * Turn all the form results into LookupResults, such that there is one lookup result
+	 * for each unique combination of stem and part-of-speech. If there are multiple lexicon entries
+	 * with the same stem and part-of-speech (eg. Judah), simply take the id of the first one
+	 * and ignore the others. This is what the Analyzer does.
 	 * 
-	 * @param {DbRowInflection} result 
-	 * @param {string} term
-	 * @returns {LookupResult}
+	 * @param {LookupResult[]} transformed_results
+	 * @param {LexicalFormResult} form_result
+	 * @returns {LookupResult[]}
 	 */
-	function transform_db_result(result, term) {
-		/** @type {string[]} */
-		// @ts-ignore
-		const form_names = result.stem === 'be'
-			? BE_FORM_NAMES
-			: FORM_NAMES.get(result.part_of_speech)
-
-		const matched_forms = result.inflections.split('|')
-			.filter(inflection => inflection !== '')
-			.map((inflection, index) => inflection.toLowerCase() === term.toLowerCase() ? index : -1)
-			.filter(index => index >= 0)
-			.map(index => form_names[index])
-			.join('|')
-
-		const form = matched_forms || 'stem'
-
-		return create_lookup_result(result, { form })
+	function transform_results(transformed_results, form_result) {
+		const existing_result = transformed_results.find(LOOKUP_FILTERS.MATCHES_LOOKUP(form_result))
+
+		if (!existing_result) {
+			// This is a new lexical entry
+			const new_result = create_lookup_result(form_result, { lexicon_id: form_result.id, form: form_result.form })
+			transformed_results.push(new_result)
+
+		} else if (existing_result.lexicon_id === form_result.id) {
+			// The lexical entry is the same as a previous one, but it has a different form. Include the new form
+			existing_result.form = `${existing_result.form}|${form_result.form.toLowerCase()}`
+
+		} else {
+			// This is a unique lexical entry with the same stem and part-of-speech as a previous one.
```

Comment:
this part is just here for the documentation?

Reply:
Yes, just to make it clear what not passing the other branch conditions means in a broader context. I tried to match the condition flow in `lookups/form.js`, `lookups/ontology.js`, and `lookups/how_to.js`